warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FieldMissingFromDateRangeVisitor.java

```diff
@@ -0,0 +1,262 @@
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyyMMdd");
+        List<ASTIdentifier> identifiers;
+
+        int numChildren = node.jjtGetNumChildren();
+
+        for (int i = 0; i < numChildren; i++) {
+            JexlNode child = node.jjtGetChild(i);
+
+            // A node could be literal == literal in terms of an identityQuery
+            try {
+                identifiers = JexlASTHelper.getIdentifiers(child);
+            } catch (NoSuchElementException e) {
+                return nonExistentFieldNames;
+            }
+
+            if (identifiers.isEmpty()) {
+                // Catch cases where we have two literals
+                // essentially everything but identifier op literal
+                return nonExistentFieldNames;
+            }
+
+            for (ASTIdentifier identifier : identifiers) {
+                String fieldName = JexlASTHelper.deconstructIdentifier(identifier);
+                if (!specialFields.contains(fieldName)) {
+                    fieldNamesToTestDateRange.add(fieldName);
+                }
+            }
+        }
+        Set<String> missingFields = helper.getMissingFieldsInDateRange(fieldNamesToTestDateRange, datatypeFilter,
+                        formatter.format(this.querySettings.getBeginDate()), formatter.format(this.querySettings.getEndDate()), specialFields);
+        if (missingFields.containsAll(fieldNamesToTestDateRange)) {
+            return nonExistentFieldNames.addAll(missingFields);
+        } else {
+            return nonExistentFieldNames;
+        }
+    }
+
+    /**
+     * @param node
+     *            Jexl node
+     * @param data
+     *            The set of names which we have determined have not been ingested during the date range.
+     * @return the updated set of names which have not been ingested during the date range.
+     */
+    private Object findMissingFields(JexlNode node, Object data) throws TableNotFoundException {
+        @SuppressWarnings("unchecked")
+        Set<String> nonIngestedFieldNames = (null == data) ? new HashSet<>() : (Set<String>) data;
+        List<ASTIdentifier> identifiers;
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyyMMdd");
+
+        // A node could be literal == literal in terms of an identityQuery
+        try {
+            identifiers = JexlASTHelper.getIdentifiers(node);
+        } catch (NoSuchElementException e) {
+            return nonIngestedFieldNames;
+        }
+
+        if (identifiers.isEmpty()) {
+            // Catch cases where we have two literals
+            // essentially everything but identifier op literal
+            return nonIngestedFieldNames;
+        }
+
+        for (ASTIdentifier identifier : identifiers) {
+            String fieldName = JexlASTHelper.deconstructIdentifier(identifier);
+            if (!specialFields.contains(fieldName)) {
+                nonIngestedFieldNames.addAll(helper.getMissingFieldsInDateRange(Set.of(fieldName), datatypeFilter,
```

Comment:
Is it worth trying to gather up the fields before we call getMissingFieldsInDateRange?  For very large queries where we repeat the same field hundreds of times, this could get quite expensive.